traversal/procedure/ProcedureEdge.java

```diff
@@ -462,9 +470,13 @@ private Sub(ProcedureVertex.Type from, ProcedureVertex.Type to, int order,
                     this.isTransitive = isTransitive;
                 }
 
-                FunctionalIterator<TypeVertex> superTypes(TypeVertex type) {
+                Seekable<TypeVertex, Order.Asc> superTypes(TypeVertex type) {
                     if (!isTransitive) return type.outs().edge(SUB).to();
-                    else return loop(type, Objects::nonNull, v -> v.outs().edge(SUB).to().firstOrNull());
+                    else {
+                        TreeSet<TypeVertex> superTypes = new TreeSet<>();
+                        loop(type, Objects::nonNull, v -> v.outs().edge(SUB).to().firstOrNull()).forEachRemaining(superTypes::add);
+                        return iterateSorted(ASC, superTypes);
```

Comment:
note: this could be the souce of the performance regression
this would be unavoidable though!